### ISSUE TITLE
EqualsVerifier should fail when equals/hashCode is directly inherited from Object

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jqno/equalsverifier#66]
```
What feature do you request?
When testing equals/hashCode it should not be allowed to use the 
equals/hashCode implementation of Object. Instead EqualsVerifier should fail 
unless equals/hashCode will be overridden

What problem will this feature solve?
This feature will solve a falsepositive where the default implementation of 
equals/hashCode returns true, but the objects actually are not true.

Please provide any additional information below.
https://groups.google.com/forum/?fromgroups=&hl=de#!topic/equalsverifier/MxuGgfg
hyE4
```

Original issue reported on code.google.com by `briem.l...@gmail.com` on 12 Sep 2012 at 6:55

**Comment by mcmoe** [comment]
Hello @jqno, 

Any plans to introduce this feature?

It would be nice to be able to add a constraint like "shouldOverride()", similar to the "allFieldsShouldBeUsed()" method.

I ran across an issue where we verified the class behavior, but the class itself did not override equals and hashcode. We would like to be able to enforce verification of the override as well.

Currently, the one way I see us enforcing a failure in such a case, is to declare which fields should be used using the allFieldsShouldBeUsed*().

**Comment by jqno** [comment]
Hi,

Yes, it's still planned. I was going to wait until I get round to version 2.0, because it would be an API breaker. I have several more plans and ideas lined up that would break the API.
But now I'm starting to think, maybe I should be a little bit less ambitious, and not try to implement all these things at the same time :).

Either way, I haven't set a date or anything. But I haven't forgotten about this either.

Jan

**Comment by mcmoe** [comment]
Thanks for the reply.

I'd like to ask, how does it break the API?

Good to know it's on your list; I Will be watching this space then :) :+1:

Cheers

**Comment by jqno** [comment]
Maybe "break the API" wasn't the best choice of words.

I want to make it easy for people to upgrade to new versions of EqualsVerifier. That means it's ok to add things or to fix bugs, but we need to be careful about changing existing behavior. I don't want people to have to change all their existing tests if they upgrade.

In this case, the number of such changes would probably be minimal (because I don't expect there will be many EV tests on classes that don't override equals), but I probably want to combine this issue with Issue #65 (make "allFieldsShouldBeUsed" the default), since these are very similar. And that definitely would cause the upgrade to be more complicated for people.

Hope this answers your question!

**Comment by mcmoe** [comment]
Ah ok makes sense if you intend to make it the default behavior.

I was thinking more on the lines of additional/optional constraint, sort of how allFieldsShouldBeUsed currently works.

Thanks for the info!

**Comment by jqno** [comment]
This has been included in version 2.0!
